# Predicting Taxi Gratuities in NYC
**Stage 3 - Statistical analysis**

**Situation:** We are reaching the midpoint of our project for the New York City Taxi & Limousine Commission (New York City TLC). We have already completed a project proposal, Python coding work, and exploratory data analysis.

Our next task is: to analyze the relationship between fare amount and payment type. A specific assignment is: to conduct an A/B test.

# Statistical analysis

In this stage, I will be using statistics to analyze and interpret our data. I'll cover fundamental concepts such as descriptive statistics and hypothesis testing. I'm going to explore the provided dataset and conduct A/B and hypothesis testing.  
<br/>   

**The purpose** of this stage is to demonstrate preparing, creating, and analyzing A/B tests. Our A/B test results aim to find ways to generate more revenue for taxi cab drivers.

For this stage, we assume that the sample data comes from an experiment in which customers were randomly selected and divided into two groups: 1) customers who are required to pay with credit card, 2) customers who are required to pay with cash. Without this assumption, we cannot draw causal conclusions about how payment method affects fare amount.

**Our goal** is to apply descriptive statistics and hypothesis testing in Python. The objective for this A/B test is to sample data and analyze whether there is a relationship between payment type and fare amount. For example, we want to discover if customers who use credit cards pay higher fare amounts than customers who use cash.
  
*Our workflow consists of four parts:*

**Part 1:** Imports and data loading

**Part 2:** EDA and hypothesis testing

**Part 3:** Communicating insights with stakeholders


# **Conducting our A/B test**

# **PACE stages**


Throughout this notebook, we make reference to the problem-solving framework PACE. I have labeled the following notebook components with their respective PACE stage: Plan, Analyze, Construct, and Execute.

## PACE: Plan

> My question is: is there an effect of payment method on fare amount?

*Let's perform the following steps to complete the statistical analysis of our data:*

### Part 1. Imports and data loading

First, we will import the packages and libraries needed to compute descriptive statistics and conduct our hypothesis test.

In [ ]:
# Importing required libraries for statistical analysis and manipulation
import pandas as pd
import numpy as np
from scipy import stats
import statsmodels as st

In [ ]:
# Loading the dataset into a pandas dataframe
data = pd.read_csv("2017_Yellow_Taxi_Trip_Data.csv", index_col = 0)

## PACE: **Analyze and Construct**

Descriptive statistics help us determine the important characeteristics of the data such as the center and dispersion which helps us determine how our data is distributed and centered.

### Part 2. Data exploration

Let's use descriptive statistics to perform our Exploratory Data Analysis (EDA).

In the dataset, `payment_type` is encoded in integers:
*   1: Credit card
*   2: Cash
*   3: No charge
*   4: Dispute
*   5: Unknown



In [ ]:
# Converting the payment type column to category and describing it, then gathering value distribution
data['payment_type'].astype('category').describe()
data['payment_type'].value_counts(normalize=True)

1    0.672497
2    0.320146
3    0.005331
4    0.002027
Name: payment_type, dtype: float64

We are interested in the relationship between how a customer pays and the fare amount they pay. Our chosen approach is to look at the average fare amount for each payment type.

In [ ]:
# Grouping by payment type and calculating the mean fare amount
data.groupby('payment_type')['fare_amount'].mean()

payment_type
1    13.429748
2    12.213546
3    12.186116
4     9.913043
Name: fare_amount, dtype: float64

Based on the averages we computed, it appears that customers who pay with a credit card tend to pay a larger fare amount than customers who pay with cash. However, this difference might arise from random sampling, rather than being a true difference in fare amount. To assess whether the difference is statistically significant, we will conduct a hypothesis test.

### Part 3. Hypothesis testing

Before conducting our hypothesis test, we should define our hypotheses:

$H_0$: There is no difference in the average fare amount between customers who use credit cards and customers who use cash.

$H_A$: There is a difference in the average fare amount between customers who use credit cards and customers who use cash.

Our goal in this step is to run a two-sample t-test. Let's recall the standard procedure:

1. State our null hypothesis and alternative hypothesis.
2. Select our significance level.
3. Compute our p-value.
4. Determine whether to reject or fail to reject the null hypothesis.

For our purposes here, this hypothesis test serves as the main component of our A/B test.

We choose 5% as our significance level and proceed to perform our two-sample t-test.

In [ ]:
# Running a two-sample t-test comparing credit card payments vs cash payments
significance_level = 0.05
stats.ttest_ind(a=data[data['payment_type']==1]['fare_amount'], b=data[data['payment_type']==2]['fare_amount'], equal_var=False)

Ttest_indResult(statistic=6.866800855655372, pvalue=6.797387473030518e-12)

> P-value = 6.8 * $10^{-12}$ , which is less than the significance level. Therefore, we `Reject` the null hypothesis

## PACE: **Execute**

Let's reflect on our Execute stage based on our findings.

### Part 4. Communicating insights with stakeholders

1. We can conclude that there is a strong relation between fare amount and payment method.
2. There might be bias in the test like the sample was taken contains much more representation of credit card payment and also it doesn't take into account no-charge or dispute cases.